File: leet_challenges/Easy/roman-to-integer/SMITH.py
class Solution:
    def romanToInt(self, s: str) -> int:
        romanDictionary = {
            "I":1,
            "V":5,
            "X":10,
            "L":50,
            "C":100,
            "D":500,
            "M":1000
        }
        count = 0
        lastNumber = 0
        
        for numeral in s:
            currentNumber = romanDictionary[numeral]
            if(currentNumber == lastNumber * 5 or currentNumber == lastNumber * 10 ):
                count += currentNumber - (lastNumber  * 2)
                lastNumber = romanDictionary[numeral]    
                continue
            count += currentNumber
            lastNumber = romanDictionary[numeral]

        return count

File: leet_challenges/Easy/roman-to-integer/SMITH.cpp
class Solution {
public:
    int switchRoman(string num,string lNum){
        if(num == "I"){
                return 1;
        }
        else if(num == "V"){
            return lNum == "I" ? 3 : 5;            
        }
        else if(num == "X"){
            return lNum == "I" ? 8 : 10;            
        }
        else if(num == "L"){
            return lNum == "X" ? 30 : 50;            
        }
        else if(num == "C"){
            return lNum == "X" ? 80 : 100;            
        }
        else if(num == "D"){
            return lNum == "C" ? 300 : 500;            
        }
        else if(num == "M"){
            return lNum == "C" ? 800 : 1000;            
        }
                
        return 0;
    }
    int romanToInt(string s) {

        int count = 0;
        string numeral = "";
        string lNum = "";
        
        for(int i = 0; i < s.length(); i++){
            numeral = s[i];
            count += switchRoman(numeral, lNum);
            lNum = numeral;
        }
        return count;
    }
};